doc/specs/stdlib_system.md

```diff
@@ -0,0 +1,427 @@
+
+`stdin` (optional): Shall be a `character(*)` value containing input to send to the process via standard input (pipe). This is an `intent(in)` argument.
+
+`want_stdout` (optional): Shall be a `logical` flag. If `.true.`, the standard output of the process will be captured; if `.false.` (default), it will be lost. This is an `intent(in)` argument.
+
+`want_stderr` (optional): Shall be a logical flag. If `.true.`, the standard error output of the process will be captured. Default: `.false.`. This is an `intent(in)` argument.
+
+`callback` (optional): Shall be a procedure conforming to the `process_callback` interface. If present, this function will be called upon process completion with the process ID, exit state, and optionally collected standard input, output, and error streams. This is an `intent(in)` argument.
+
+`payload` (optional): Shall be a generic (`class(*)`) scalar that will be passed to the callback function upon process completion. It allows users to associate custom data with the process execution. This is an `intent(inout), target` argument.
+
+### Return Value
+
+Returns an object of type `process_type` that contains information about the state of the created process.
+
+### Example
+
+```fortran
+! Example usage with command line or list of arguments
+type(process_type) :: p(2)
+
+! Run a simple command line asynchronously
+p(1) = runasync("echo 'Hello, world!'", want_stdout=.true.)
+
+! Run a command using an argument list asynchronously
+p(2) = runasync(["/usr/bin/ls", "-l"], want_stdout=.true.)
+```
+
+## `is_running` - Check if a process is still running
+
+### Status
+
+Experimental
+
+### Description
+
+The `is_running` interface provides a method to check if an external process is still running.  
+This is useful for monitoring the status of asynchronous processes created with the `run` interface.
+
+### Syntax
+
+`status = ` [[stdlib_subprocess(module):is_running(interface)]] `(process)`
+
+### Arguments
+
+`process`:  Shall be a `type(process_type)` object representing the external process to check. This is an `intent(inout)` argument.
+
+
+### Return Value
+
+Returns a `logical` value: `.true.` if the process is still running, or `.false.` if the process has terminated.
+After a call to `is_running`, the `type(process_type)` structure is also updated to the latest process state.
+
+### Example
+
+```fortran
+! Example usage of is_running
+type(process_type) :: proc
+logical :: status
+
+! Start an asynchronous process
+proc = run("sleep 10", wait=.false.)
+
+! Check if the process is running
+status = is_running(proc)
+
+if (status) then
+    print *, "Process is still running."
+else
+    print *, "Process has terminated."
+end if
+```
+
+## `is_completed` - Check if a process has completed execution
```

Comment:
What is the addition of `is_completed` in comparison to `is_running`? It seems like the result will always be `is_completed(process) == .not.is_running(process)`?